src/TraceEvent/Parsers/ClrTraceEventParser.cs

```diff
@@ -2641,16 +2771,198 @@ public override object PayloadValue(int index)
+        /// The counting has a couple of heuristics that avoid spending too much time counting.
+        /// </summary>
+        LargestFreeListItems = 0,
+
+        /// <summary>
+        /// When the BucketKind is PlugsInCondemned, the buckets in the associated GCFitBucketInfo events
+        /// represents plugs that we found during the plan phase of a Gen1 GC where it is planned to fit in the 
+        /// current (i.e. condemned) generation. 
+        /// </summary>
+        PlugsInCondemned = 1
+    }
+
+    public class GCFitBucketInfoTraceData : TraceEvent
+    {
+        public int ClrInstanceID { get { return GetInt16At(0); } }
+
+        /// <summary>
+        /// The kind of the buckets (see the comments for the enum BucketKind for more information)
+        /// </summary>
+        public BucketKind BucketKind { get { return (BucketKind)GetInt16At(2); } }
+
+        /// <summary>
+        /// The total size of all buckets.
+        /// </summary>
+        public long TotalSize { get { return GetInt64At(4); } }
+
+        /// <summary>
+        /// The number of buckets.
+        /// </summary>
+        public int Count { get { return GetInt16At(12); } }
+
+        public GCFitBucket Buckets(int bucketIndex)
+        {
+            Debug.Assert(0 <= bucketIndex && bucketIndex < Count);
+            int baseAddress = 14 + SizeOfGCFitBucket * bucketIndex;
+            int index = GetInt32At(baseAddress);
+            int count = GetInt32At(baseAddress + 4);
+            long size = (PointerSize == 8) ? GetInt64At(baseAddress + 8) : (long)GetInt32At(baseAddress + 8);
+            return new GCFitBucket(index, count, size);
+        }
+
+
+        #region Private
+        internal GCFitBucketInfoTraceData(Action<GCFitBucketInfoTraceData> action, int eventID, int task, string taskName, Guid taskGuid, int opcode, string opcodeName, Guid providerGuid, string providerName)
+            : base(eventID, task, taskName, taskGuid, opcode, opcodeName, providerGuid, providerName)
+        {
+            Action = action;
+        }
+        protected internal override void Dispatch()
+        {
+            Action(this);
+        }
+        protected internal override Delegate Target
+        {
+            get { return Action; }
+            set { Action = (Action<GCFitBucketInfoTraceData>)value; }
+        }
+        protected internal override void Validate()
+        {
+            int size0 = 14 + Count * SizeOfGCFitBucket;
+            Debug.Assert(!(Version == 0 && EventDataLength != size0));
+            Debug.Assert(!(Version > 0 && EventDataLength < size0));
+        }
+        public override StringBuilder ToXml(StringBuilder sb)
+        {
+            Prefix(sb);
+            XmlAttrib(sb, "ClrInstanceID", ClrInstanceID);
+            XmlAttrib(sb, "BucketKind", BucketKind);
+            XmlAttrib(sb, "TotalSize", TotalSize);
+            XmlAttrib(sb, "Count", Count);
```

Comment:
I would expect buckets to be written out here as well.

Reply:
@brianrob, but `Bucket` is an array of struct, how exactly will we do that?

Reply:
My thought is that you'd need to write some code to cast each of these entries to the struct, and then write the struct out in XML.  This stuff tends to be useful when you don't trust what's in the events, or want to dump their values into XML in the PerfView log.

Reply:
Got it, I think we can do something along the line of `GCBulkTypeTraceData.ToXml()`

Reply:
Yes, that sounds great.